docs/release-info/release-notes.md

```diff
@@ -57,63 +57,373 @@ For tips about how to write a good release note, see [Release notes](https://git
+
+Jetty 12 servers  do strict SNI `host0` checks when TLS is enabled. If the host your client is connecting to the server with does not match what is in the keystore, even if there is only one certificate in that keystore, it will return a 400 response. This could impact some use cases, such as folks connecting over `localhost` for whatever reason. If this change will break your deployment, you can opt-out of the change by setting `druid.server.http.enforceStrictSNIHostChecking` to false in the runtime.properties for some or all of your Druid services. It is recommended that you modify your client behavior to accommodate this change in `jetty` instead of overriding the config whenever possible.
+
+[#18424](https://github.com/apache/druid/pull/18424) [#18623](https://github.com/apache/druid/pull/18623)
+
+#### Kerberos authentication
+
+The `druid.auth.authenticator.kerberos.cookieSignatureSecret` config is now mandatory. 
+
+[#18368](https://github.com/apache/druid/pull/18368)
 
-<!--Carry this forward until 32. Then move it to incompatible changes -->
+#### Multi-stage query task engine
 
-In Druid 32.0.0, the front coded dictionaries feature will be turned on by default. Front-coded dictionaries reduce storage and improve performance by optimizing for strings where the front part looks similar.
+The MSQ task engine is now a core capability of Druid rather than an extension. It has been in the default extension load list for several releases. 
 
-Once this feature is on, you cannot easily downgrade to an earlier version that does not support the feature. 
+Remove `druid-multi-stage-query` from `druid.extensions.loadList` in `common.runtimes.properties` before you upgrade.
 
-For more information, see [Migration guide: front-coded dictionaries](./migr-front-coded-dict.md).
+Druid 35.0.0 will ignore the extension if it's in the load list. Future versions of Druid will fail to start since it cannot locate the extension.
 
-If you're already using this feature, you don't need to take any action. 
+[#18394](https://github.com/apache/druid/pull/18394)
 
+#### pac4j extension
+
+Due to the upgrade from `pac4j` 4 to 5, session serialization has changed from `pac4j`’s `JavaSerializer` to standard Java serialization. As a result, clients of clusters using the `pac4j` extension may be logged out during rolling upgrades and need to re‑authenticate.
 
 ### Incompatible changes
 
+#### Java 11 support removed
+
+Upgrade to Java 17 or 21. Note that some versions of Java 21 encountered issues during test, specifically Java 21.05-21.07. If possible, avoid these versions.
+
+[#18424](https://github.com/apache/druid/pull/18424)
+
+
 ### Developer notes
```

Comment:
A note about the new embedded tests in this section would be nice too.

https://github.com/apache/druid/pull/18147